src/components/transport_manager/src/bluetooth/bluetooth_device_scanner_win.cc

```diff
@@ -451,5 +452,11 @@ void BluetoothDeviceScanner::BluetoothDeviceScannerDelegate::threadMain() {
   scanner_->Thread();
 }
 
+void BluetoothDeviceScanner::BluetoothDeviceScannerDelegate::exitThreadMain() {
+  scanner_->shutdown_requested_ = true;
```

Comment:
@pestOO According to the documentation of Microsoft Bluetooth - it is guaranteed.